clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,397 @@
+		summary, err := npm.gather()
+		if err != nil {
+			return nil, err
+		}
+		return []measurement.Summary{summary}, err
+	default:
+		return nil, fmt.Errorf("unknown action: %v", action)
+	}
+}
+
+func (npm *networkPerformanceMeasurement) start(config *measurement.Config) error {
+	npm.initialize(config)
+	if err := npm.prepareCluster(); err != nil {
+		return err
+	}
+	if err := npm.createAndWaitForWorkerPods(); err != nil {
+		return err
+	}
+	if err := npm.storeWorkerPods(); err != nil {
+		return err
+	}
+	if err := npm.initializeInformer(); err != nil {
+		return err
+	}
+
+	switch npm.podRatioType {
+	case oneToOne, manyToMany:
+		npm.execNToMTest()
+	default:
+		return fmt.Errorf("invalid Pod Ratio: %v", npm.podRatioType)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) initialize(config *measurement.Config) {
+	npm.k8sClient = config.ClusterFramework.GetClientSets().GetClient()
+	npm.framework = config.ClusterFramework
+	npm.dynamicClient = config.ClusterFramework.GetDynamicClients().GetClient()
+	npm.podInfo.workerPodMap = make(map[string]podList)
+}
+
+func (npm *networkPerformanceMeasurement) initializeInformer() error {
+	gvr, _ := meta.UnsafeGuessKindToResource(crGvk)
+	npm.resourceInterface = npm.dynamicClient.Resource(gvr).Namespace(netperfNamespace)
+	informer, err := getInformer(netperfNamespace, npm.dynamicClient, gvr)
+	if err != nil {
+		return fmt.Errorf("error getting informer:%s", err)
+	}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			npm.handleUpdateNetworkTestEvents(newObj)
+		},
+	})
+	npm.stopCh = make(chan struct{})
+	go informer.Run(npm.stopCh)
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) prepareCluster() error {
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+	if err := npm.framework.ApplyTemplatedManifests(clusterRoleBindingFilePath, nil); err != nil {
+		return fmt.Errorf("error while creating clusterRoleBinding: %v", err)
+	}
+	if err := npm.framework.ApplyTemplatedManifests(crdManifestFilePath, nil); err != nil {
+		return fmt.Errorf("error while creating CRD: %v", err)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) cleanupCluster() {
+	if err := npm.framework.DeleteObject(crdGvk, "", customResourceDefinitionName); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+	if err := npm.framework.DeleteObject(rbacGvk, "", rbacName); err != nil {
+		klog.Errorf("error while deleting clusterRoleBinding: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
```

Comment:
Waiting for namespace deletion failed: %v, err

Reply:
modified